### ISSUE TITLE
What if comparing RFTT with ReFT+RL?

### ISSUE THREAD

**Issue by john1226966735** [sastpg/RFTT#2]
Hi, Thank you for sharing this interesting and insightful work! I appreciate the detailed analysis and the novel approach you introduced.  

While reviewing the experimental results, I noticed a potential issue with the comparison setup. Specifically, I believe the paper should verify that **reasoning with functional tokens** performs better than reasoning without functional tokens by ensuring both follow the same training paradigm. From my perspective, if **reasoning with functional tokens** is indeed more effective, then in Table 1, "SFT Warmup (Ours)" should surpass ReFT, since ReFT also only use SFT to train. However, the results do not reflect this expectation.

Moreover, if we aim to rigorously verify the effectiveness of **reasoning with functional tokens**, the appropriate comparison should be between RFTT and ReFT+RL rather than ReFT alone. This would provide a more controlled evaluation of the impact of functional tokens.  

If the paper already addresses this point or includes such experiments, I might have overlooked them—apologies for any oversight on my part. I would love to hear your thoughts on this and whether you plan to explore such a comparison in future experiments.

**Comment by sastpg** [comment]
Wow! Thank you for your careful reading. I would like to further clarify the baseline setting of ReFT [1].

ReFT is also a reinforced fine-tuning method using the pipeline of SFT+RL. We follow this pattern to conduct the experiments of ReFT:
1. SFT using CoT data w/o functional token
2. RL using random sampling w/o functional token using rule-based rewards

Therefore, we think the comparison is fair.

For initial SFT data in ReFT, we don't use the functional tree search as stated in our paper. Instead, we prompt GPT-4o/Qwen-max to generate detailed step-by-step reasoning steps referring to the original short solutions provided by MATH dataset.

The code for generation CoT data in ReFT:

```
def get_response(args):
    question, expected_answer = args
    # Set OpenAI's API key and API base to use vLLM's API server.
    gen_cot_prompt = f"Given a math problem and a concise expected answer, please rewrite the solution in a step-by-step thought process. Expand the answer by providing detailed explanations for each step. Here is the question:\n{question}\n\nExpected answer:{expected_answer}\n"
    chat_response = call_api(gen_cot_prompt)
    info = json.dumps({"instruction": question, "input": "", "output":chat_response, "golden": expected_answer}, ensure_ascii=False)
    return info
```

I hope this can solve your concerns. If you have any questions, please feel free to discuss. Have a nice day :)

----
[1] REFT: Reasoning with REinforced Fine-Tuning. https://arxiv.org/pdf/2401.08967

**Comment by john1226966735** [comment]
Thank you for your clarification! I realize now that I misunderstood the meaning of ReFT—I had mistakenly assumed it referred to Rejection sampling Fine-Tuning (RFT). I sincerely apologize for not reading the paper carefully enough.

I truly appreciate your patience in addressing my question, and I remain impressed by the insights and contributions of this work. Looking forward to seeing its impact on the field!
